apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -319,79 +151,14 @@ Gallery.prototype = {
     return this.client.findElement(Gallery.Selector.openSaveButton);
   },
 
-  /**
-   * Read the translateX style and return its integer value.
-   */
-  getFrameTranslation: function(frame) {
-    var style = frame.getAttribute('style');
-    return parseInt(style.match(/.*:\s.*\((\d*).*/)[1]);
-  },
-
-  waitFor: function(selector) {
-    return this.client.helper.waitForElement(selector);
-  },
-
-  /**
-  * Wait for the image editor view to render before continuing with the tests.
-  */
-  waitForImageEditor: function() {
-    this.waitFor(Gallery.Selector.editCropButton);
-    this.waitFor(Gallery.Selector.editEffectButton);
-    this.waitFor(Gallery.Selector.editExposureButton);
-    this.waitFor(Gallery.Selector.editEnhanceButton);
-  },
-
-  enterMainEditScreen: function() {
+  tapFirstThumbnail: function() {
     this.thumbnail.click();
-    this.editButton.click();
-  },
-
-  tapCancel: function() {
-    this.waitFor(Gallery.Selector.editHeader).tap(25, 25);
-  },
-
-  getExposureSliderPosition: function() {
-    return this.exposureSlider.text();
-  },
-
-  applyEditToolOptions: function() {
-    this.editToolApplyButton.click();
-  },
-
-  saveEditedImage: function() {
-    this.editSaveButton.click();
-  },
-
-  waitForCropAspectPortraitSelected: function() {
-    this.client.waitFor(function() {
-      return this.editCropAspectPortraitButton
-                        .getAttribute('class').indexOf('selected') > -1;
-    }.bind(this));
+    return new Fullscreen_View(this.client);
```

Comment:
This is going to create new instance of fullscreenview every time user taps first thumbnail. As suggested above we should require(./fullscreen_view.js) in tests which need fullscreen view and instantiate fullscreen view once in setup.